dso-l2/src/main/java/com/tc/objectserver/entity/ActiveToPassiveReplication.java

```diff
@@ -239,13 +247,19 @@ public ActivePassiveAckWaiter replicateActivity(SyncReplicationActivity activity
       //  command.
       boolean isLocalFlush = (SyncReplicationActivity.ActivityType.FLUSH_LOCAL_PIPELINE == activity.getActivityType());
       for (NodeID node : copy) {
-        // This is a normal completion.
-        boolean didSend = false;
         if (!isLocalFlush) {
+          BarrierCompletion barrierCompletion = new BarrierCompletion();
+          Consumer<Boolean> consumer = (didSend) -> {
+            if(!didSend) {
+              boolean isNormalComplete = true;
+              internalAckCompleted(activityID, node, null, isNormalComplete);
+            }
+            barrierCompletion.complete();
+          };
           // This isn't local-only so try to replicate.
-          didSend = this.replicationSender.replicateMessage(node, activity);
-        }
-        if (!didSend) {
+          this.replicationSenderMessageSink.addSingleThreaded(new SyncActivityMessage(node, activity, consumer));
```

Comment:
We don't want to do this here.  This will slow replication way down.  What is the purpose of waiting for completion?